src/custom/state/claim/hooks/index.ts

```diff
@@ -8,12 +18,79 @@ export type ClaimType =
   | 'UserOption' // paid, with vesting, must use Native currency, can be available on both mainnet and gchain
   | 'Investor' // paid, with vesting, must use USDC, only on mainnet
 
+export const FREE_CLAIM_TYPES: ClaimType[] = ['Airdrop', 'Team', 'Advisor']
+export const PAID_CLAIM_TYPES: ClaimType[] = ['GnoOption', 'UserOption', 'Investor']
+
 export interface UserClaimData {
   index: number
   amount: string
   proof: string[]
   type: ClaimType
-  // TODO: Either add the missing fields, or add https://github.com/gnosis/gp-v2-token type
 }
 
+type Account = string | null | undefined
+
 export type UserClaims = UserClaimData[]
+
+/**
+ * Gets an array of available claim
+ *
+ * @param account
+ */
+export function useUserAvailableClaims(account: Account): UserClaims {
+  const userClaims = useUserClaims(account)
+  const contract = useVCowContract()
+
+  // build list of parameters, with the claim index
+  const claimIndexes = userClaims?.map(({ index }) => [index]) || []
+
+  const results = useSingleContractMultipleData(contract, 'isClaimed', claimIndexes)
+
+  console.log(`useUserAvailableClaims::re-render`, userClaims, claimIndexes, results)
+
+  return useMemo(() => {
+    if (!userClaims || userClaims.length === 0) {
+      // user has no claims
+      return []
+    }
+
+    return results.reduce<UserClaims>((acc, result, index) => {
```

Comment:
i think this would be simpler as a filter

Reply:
But then I have to filter and map, while on reduce it's a single operation, no?

Of course, it has negligible performance impact. Will refactor.

Reply:
Actually, if I filter then map I'll lose the reference of the index.
If I map then filter I'll have to build another temporary data structure which is very similar to the current one.

I'd like to keep it as is if you don't mind